File: PRNumber/app/src/main/java/com/pplus/prnumberuser/core/network/upload/S3Upload.java
package com.pplus.prnumberuser.core.network.upload;

import com.google.gson.reflect.TypeToken;
import com.pplus.prnumberuser.Const;
import com.pplus.prnumberuser.core.network.model.dto.Attachment;
import com.pplus.prnumberuser.core.network.model.response.NewResultResponse;

/**
 * Created by j2n on 2016. 9. 30..
 */

public class S3Upload extends AbstractUpload<Attachment>{

    public S3Upload(PplusUploadListener<Attachment> pplusCallback){

        super(pplusCallback);
    }

    @Override
    String getUploadUrl(){

        return Const.API_URL + "common/saveOnlyS3";
    }

    @Override
    boolean isMultiThreadEnable(){

        return true;
    }

    @Override
    TypeToken<NewResultResponse<Attachment>> getResultType(){

        return new TypeToken<NewResultResponse<Attachment>>(){};
    }
}

File: Buflexz/app/src/main/java/com/root37/buflexz/core/network/upload/S3Upload.kt
package com.root37.buflexz.core.network.upload

import com.google.gson.reflect.TypeToken
import com.root37.buflexz.Const
import com.root37.buflexz.core.network.model.response.NewResultResponse

/**
 * Created by j2n on 2016. 9. 30..
 */
class S3Upload(callback: PplusUploadListener<String>) : AbstractUpload<String>(callback) {


    override fun getUploadUrl(): String {
        return Const.API_URL + "file/s3Upload"
    }

    override fun isMultiThreadEnable(): Boolean {
        return true
    }

    override fun getResultType(): TypeToken<NewResultResponse<String>> {
        return object : TypeToken<NewResultResponse<String>>() {}
    }
}